src/main/java/com/google/starfish/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,195 @@
+package com.google.starfish;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.sql.DataSource;
+
+@WebListener("Creates a connection pool that is stored in the Servlet's context for later use.")
+public class ConnectionPoolContextListener implements ServletContextListener {
+
+  // All database credentials are stored in environment variables
+  private static final String CLOUD_SQL_CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASS = System.getenv("DB_PASS");
+  private static final String DB_NAME = System.getenv("DB_NAME");
+
+  private final int MILLISECONDS_IN_SECOND = 1000;
+  private final int SECONDS_IN_MINUTE = SECOND * 60;
+
+  /** Configures database connection, opens pool, and returns pool as DataSource. */
+  private DataSource createConnectionPool() {
+    // The configuration object specifies behaviors for the connection pool.
+    HikariConfig config = new HikariConfig();
```

Comment:
I'm not sure what Hikari is. Is it some standard way of setting up JDBC config files?

Reply:
Basically, but in a more general sense it's straight up a no assembly required JDBC connection pool. You can read more about it on [the official github page](https://github.com/brettwooldridge/HikariCP).

Reply:
Interesting, is it something you've used before or that was recommended by the CloudSQL documentation? Don't mean to pry, sorry 😛

Reply:
I have seen it before but haven't had the opportunity to use it since this is the first time I'm doing CloudSQL stuff in Java. Although it wasn't explicitly mentioned in the CloudSQL documentation, there's a Google Cloud github page where they have examples and I did see them using Hikari to set the config so it seems like it's pretty standard.

Reply:
Very cool, good to know